editoast/editoast_derive/src/snapshots/editoast_derive__error__tests__construction.snap

```diff
@@ -0,0 +1,71 @@
+---
+source: editoast_derive/src/error.rs
+---
+// Source
+// -----------------------------------------------------------------------------
+
+#[derive(EditoastError)]
+#[editoast_error(base_id = "infra", default_status = 500)]
+pub enum InfraApiError {
+    #[editoast_error(status = 404, no_context)]
+    NotFound { infra_id: i64 },
+    #[editoast_error(status = 400)]
+    BadRequest { message: String },
+    InternalError,
+}
+
+// Macro expansion
+// -----------------------------------------------------------------------------
+
+inventory::submit! {
+    crate ::error::ErrorDefinition::new("editoast:infra:NotFound", "NotFound",
+    "InfraApiError", 404u16, "{\"infra_id\":\"i64\"}")
+}
+inventory::submit! {
+    crate ::error::ErrorDefinition::new("editoast:infra:BadRequest", "BadRequest",
+    "InfraApiError", 400u16, "{\"message\":\"String\"}")
+}
+inventory::submit! {
+    crate ::error::ErrorDefinition::new("editoast:infra:InternalError", "InternalError",
+    "InfraApiError", 500u16, "{}")
+}
+impl crate::error::EditoastError for InfraApiError {
+    fn get_status(&self) -> axum::http::StatusCode {
+        match self {
+            #[allow(unused)]
+            Self::NotFound { infra_id } => {
+                axum::http::StatusCode::try_from(404)
+                    .expect("EditoastError: invalid status expression")
+            }
+            #[allow(unused)]
+            Self::BadRequest { message } => {
+                axum::http::StatusCode::try_from(400)
+                    .expect("EditoastError: invalid status expression")
+            }
+            #[allow(unused)]
+            Self::InternalError => axum::http::StatusCode::from_u16(500u16).unwrap(),
```

Comment:
Unrelated to current PR: Interesting to see that sometimes, it's expanded to `StatusCode::try_from(400)` and sometimes to `StatusCode::from_u16(500u16)`. We might want to make it the same? Or do I miss a reason why it's different?

Reply:
Not sure why it's different. We could correct that in another PR but since this macro will be removed sooner than later, idk 🤷